### DISCUSSION TITLE
Visual Studio 2022 extension

### DISCUSSION THREAD

**Discussion by llfab** [AvaloniaUI/Avalonia#7178] [Q&A] [answered]
I have walked through some other discussions about the VS 2022 Avalonia extension. There seems some work going on and some sort of alpha version. I wonder whether there is already a release date / plan for this?

**Comment by Gillibald** [comment, marked as answer]
https://marketplace.visualstudio.com/items?itemName=AvaloniaTeam.AvaloniaVS

**Reply by llfab** [reply]
Uuups. Thanks. I really did not check for a couple of days and just read all the other threats last week :-)

**Comment by KarlCHansen** [comment]
The latest (V1.68.1) version of VSCode **_refuses_** to install the Avalonia extension that is supposed to be for VSCode 2022.

VSCode throws the error "extension/package.json not found inside zip".

I find this rather USELESS unless it gets fixed ASAP.

**Reply by KarlCHansen** [reply]
The previous test was on Linux, Ubuntu 20.04.  I will test on Windows 10.

**Reply by jp2masa** [reply]
This discussion is about VS 2022, not VS Code.

**Reply by KarlCHansen** [reply]
...and *MOST* VS extensions also work in VSCode.  Furthermore, because VS is a pay-to-play platform, and there are thousands more who use VSCode, it makes no sense to support *only* the pay-to-play platform.

**Reply by maxkatz6** [reply]
Don't get confused. 
VS and VSCode have quite different extensions API. Furthermore, even for Visual Studio itself it's not always possible to get "single package" for 2019 and 2022 version (thus we have this discussion created, because 2019 extension didn't work for 2022, and now we have TWO of them), and you even ask for VSCode at the same time.
Not to mention, in .NET ecosystem VS and Rider are way more used comparing to VSCode. We might like it or not, but that's how it is in .NET ecosystem.

**Reply by maxkatz6** [reply]
Saying that, if you want VSCode extension specifically, you should take a look at this repository https://github.com/AvaloniaUI/Avalonia.VSCodeExtension

It doesn't have full functionality, for instance there is no previewer yet. But code competition and syntax highlighting works.

This discussion is Visual Studio specific, and I am going to close it. 
Thanks.

**Comment by KarlCHansen** [comment]
Tested on Win10. Same results.

The latest (V1.68.1) version of VSCode refuses to install the Avalonia extension that is supposed to be for VSCode 2022.

VSCode throws the error "extension/package.json not found inside zip".

I find this rather USELESS unless it gets fixed ASAP.
